app/feature/feature-changeaddress/src/main/kotlin/com/hedvig/android/feature/changeaddress/destination/enternewaddress/EnterNewAddressViewModel.kt

```diff
@@ -0,0 +1,282 @@
+          currentState = currentState.copy(
+            navParamsForVillaDestination = null,
+            navParamsForOfferDestination = null,
+          )
+        }
+      }
+    }
+
+    return currentState
+  }
+}
+
+internal data class EnterNewAddressUiState(
+  val isLoading: Boolean = false,
+  val errorMessage: String? = null,
+  val street: ValidatedInput<String?> = ValidatedInput(null),
+  val postalCode: ValidatedInput<String?> = ValidatedInput(null),
+  val squareMeters: ValidatedInput<String?> = ValidatedInput(null),
+  val numberInsured: ValidatedInput<String?> = ValidatedInput(null),
+  val datePickerUiState: DatePickerUiState? = null,
+  val isEligibleForStudent: Boolean = false,
+  val isStudent: Boolean = false,
+  val movingDate: ValidatedInput<LocalDate?> = ValidatedInput(null),
+  val maxNumberCoInsured: Int? = null,
+  val maxSquareMeters: Int? = null,
+  val navParamsForVillaDestination: MovingParameters? = null,
+  val navParamsForOfferDestination: MovingParameters? = null,
+) {
+  val isAddressInputValid: Boolean
+    get() {
+      return street.errorMessageRes == null &&
+        postalCode.errorMessageRes == null &&
+        squareMeters.errorMessageRes == null &&
+        isSquareMetersWithinBounds(squareMeters.input?.toIntOrNull()) &&
+        movingDate.errorMessageRes == null &&
+        numberInsured.errorMessageRes == null
+    }
+
+  fun validateAddressInput(): EnterNewAddressUiState {
+    return copy(
+      street = street.copy(
+        errorMessageRes = if (!street.isPresent || street.input?.isBlank() == true) {
+          hedvig.resources.R.string.CHANGE_ADDRESS_STREET_ERROR
+        } else {
+          null
+        },
+      ),
+      postalCode = postalCode.copy(
+        errorMessageRes = if (!postalCode.isPresent || postalCode.input?.isBlank() == true) {
+          hedvig.resources.R.string.CHANGE_ADDRESS_POSTAL_CODE_ERROR
+        } else {
+          null
+        },
+      ),
+      squareMeters = squareMeters.copy(
+        errorMessageRes = if (!squareMeters.isPresent || squareMeters.input?.isBlank() == true) {
+          hedvig.resources.R.string.CHANGE_ADDRESS_LIVING_SPACE_ERROR
+        } else if (squareMeters.isPresent && !isSquareMetersWithinBounds(squareMeters.input!!.toIntOrNull())) {
+          hedvig.resources.R.string.CHANGE_ADDRESS_LIVING_SPACE_OVER_LIMIT_ERROR
+        } else {
+          null
+        },
+      ),
+      movingDate = movingDate.copy(
+        errorMessageRes = if (!movingDate.isPresent) {
+          hedvig.resources.R.string.CHANGE_ADDRESS_MOVING_DATE_ERROR
+        } else {
+          null
+        },
+      ),
+      numberInsured = numberInsured.copy(
+        errorMessageRes = if (!numberInsured.isPresent || isNumberInsuredTooLow(numberInsured.input!!.toIntOrNull())) {
+          hedvig.resources.R.string.CHANGE_ADDRESS_CO_INSURED_ERROR // todo: I think the copy may be wrong here. It is the number of all the insured (incl.the main insured, not only co-insured)
```

Comment:
Can we ask iOS too what they do?

Reply:
They actually have a completely different field here:) which I think is better (easier to understand)

<img src="https://github.com/HedvigInsurance/android/assets/90948269/82f1f043-513f-4cdb-bb17-8e4dc20b7b96" width="252" height="533">

Reply:
Do you we don't have this at all in Android I think right?
New design I assume has this component?
Also, do we have a new design for moving flow anyway?
Maybe this could be fixed while those are being worked on so we don't make smth now only to be replaced soon. Otherwise if new design does not have moving flow, perhaps we should just make a quick impl for this which fixes it.

Reply:
Kinda did a quick impl without checking:))) but we do in fact have this component in the new design: https://www.figma.com/design/5kmmDdh6StpXzbEfr7WevV/Hedvig-UI-Kit?node-id=15724-23348&t=uKXQD6goM9XLlf5C-4

Reply:
But I say it's a whole new component, so would take some time to implement. Maybe a separate PR for this?